src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -465,6 +465,14 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
 
     var value;
 
+    const classWithRenderNotExtended = !(workInProgress.type.prototype &&
+      workInProgress.type.prototype.render);
+    invariant(
+      classWithRenderNotExtended,
+      'Using a class with a render method as a function, ' +
+        'did you forget to extend React.Component?',
+    );
```

Comment:
Let's make this a warning instead, so that we can strip this out in production.

```js
if (__DEV__) {
  warning(workInProgress.type.prototype &&
 +      workInProgress.type.prototype.render, '...');
}
```

Reply:
thanks sebastian - just doublechecking before I fix - the issue discussion last left saying it should be an invariant outside a `__DEV__` block ([see here](https://github.com/facebook/react/issues/10103#issuecomment-313245162)) - are you sure you want it just as a warning?

Reply:
regardless, i will resubmit as requested :) really honored to help.